articles/django-crud-api/index.md

```diff
@@ -0,0 +1,320 @@
+
+### Creating the TodoAPIView
+In the `views.py` file in our `todo` app directory, we are going to write the logic for the CRUD functionality for our app. Django Rest framework comes with inbuilt classes which make building the CRUD functionality very easy. We start by importing the Django rest framework classes that we will use to create our CRUD API. We just need to create a class for each of our crud endpoints and add in the `queryset` and the `serializer_class` for our `Todo` model.
+
+```python
+from django.shortcuts import render
+from rest_framework.generics import ListAPIView
+from rest_framework.generics import CreateAPIView
+from rest_framework.generics import DestroyAPIView
+from rest_framework.generics import UpdateAPIView
+from todo.serializers import TodoSerializer
+from todo.models import Todo
+
+# Create your views here.
+class ListTodoAPIView(ListAPIView):
+    """This endpoint list all of the available todos from the database"""
+    queryset = Todo.objects.all()
+    serializer_class = TodoSerializer
+
+class CreateTodoAPIView(CreateAPIView):
+    """This endpoint allows for creation of a todo"""
+    queryset = Todo.objects.all()
+    serializer_class = TodoSerializer
+
+class UpdateTodoAPIView(UpdateAPIView):
+    """This endpoint allows for updating a specific todo by passing in the id of the todo to update"""
+    queryset = Todo.objects.all()
+    serializer_class = TodoSerializer
+
+class DeleteTodoAPIView(DestroyAPIView):
+    """This endpoint allows for deletion of a specific Todo from the database"""
+    queryset = Todo.objects.all()
+    serializer_class = TodoSerializer
+```
+### Creating URL paths for our Todo endpoints.
+The URLs allows us to interact with the various `Todo` crud views. In the `todo` app directory create a file `urls.py` where we will write URLs for various endpoints.
+
+```python
+from django.urls import path
+from todo import views
+
+urlpatterns = [
+    path("",views.ListTodoAPIView.as_view(),name="todo_list"),
+    path("create/", views.CreateTodoAPIView.as_view(),name="todo_create"),
+    path("update/<int:pk>/",views.UpdateTodoAPIView.as_view(),name="update_todo"),
+    path("delete/<int:pk>/",views.DeleteTodoAPIView.as_view(),name="delete_todo")
+]
+```
+In the `urls.py` in the `django_todo` project directory, let's added the base URL for our app so that the `django_todo` project can be aware of the `todo` app URLs.
+
+```python
+from django.contrib import admin
+from django.urls import path
+from django.urls import include
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('api/v1/todo/',include("todo.urls"))
+]
+```
+### Testing the endpoints
+
+Making a GET request to http://localhost:8000/api/v1/todo in postman returns a list of `todos`.
```

Comment:
I feel like you should point out that the server should be running.
Consider addding an instruction to run the command that runs the server.